{{cookiecutter.python_name}}/src/index.ts

```diff
@@ -1,18 +1,32 @@
 import {
   JupyterFrontEnd,
   JupyterFrontEndPlugin
-} from '@jupyterlab/application';{% if cookiecutter.has_server_extension.lower().startswith('y') %}
+} from '@jupyterlab/application';{% if cookiecutter.has_settings.lower().startswith('y') %}
+
+import { ISettingRegistry } from '@jupyterlab/settingregistry';{% endif %}{% if cookiecutter.has_server_extension.lower().startswith('y') %}
 
 import { requestAPI } from './handler';{% endif %}
 
 /**
  * Initialization data for the {{ cookiecutter.labextension_name }} extension.
  */
-const extension: JupyterFrontEndPlugin<void> = {
+const plugin: JupyterFrontEndPlugin<void> = {
   id: '{{ cookiecutter.labextension_name }}:plugin',
-  autoStart: true,
-  activate: (app: JupyterFrontEnd) => {
-    console.log('JupyterLab extension {{ cookiecutter.labextension_name }} is activated!');{% if cookiecutter.has_server_extension.lower().startswith('y') %}
+  autoStart: true,{% if cookiecutter.has_settings.lower().startswith('y') %}
+  optional: [ISettingRegistry],{% endif %}
+  activate: (app: JupyterFrontEnd{% if cookiecutter.has_settings.lower().startswith('y') %}, settingRegistry: ISettingRegistry | null{% endif %}) => {
+    console.log('JupyterLab extension {{ cookiecutter.labextension_name }} is activated!');{% if cookiecutter.has_settings.lower().startswith('y') %}
+
+    if(settingRegistry){
```

Comment:
Maybe this was one of the lint issues:

```suggestion
    if (settingRegistry) {
```

Reply:
Thanks @jtpio. Indeed that was part of the error (I correct it). But the linter is also complaining on the `activate` signature. If the setting registry is present, the two arguments needs to be put in separated lines. But with no settings, `app` argument is required to be inline. It seems unnecessary to add complexity to the template so that both cases pass the lint test.